dfu: dfu_target: Reset a properly downloaded image on dfu_target_reset() (#9304)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

See the changelog for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for additional information.

DFU target
----------

* Updated:

>   * Calling :c:func:`dfu_target_reset()` will clear any image that is already downloaded into a target area. This allows cancelling any update packages even if they are already marked to be updated.

Reviewer: ```suggestion
  * Calling the :c:func:`dfu_target_reset()` function clears all images that have already been downloaded into a target area.
     This allows cancelling any update packages even if they are already marked to be updated.
```

Reviewer: Not done properly yet. All sentences must be on separate lines.
